k8-operator/controllers/infisicalsecret_helper.go

```diff
@@ -239,41 +291,65 @@ func (r *InfisicalSecretReconciler) ReconcileInfisicalSecret(ctx context.Context
 		secretVersionBasedOnETag = managedKubeSecret.Annotations[SECRET_VERSION_ANNOTATION]
 	}
 
+	if authStrategy == AuthStrategy.UNIVERSAL_MACHINE_IDENTITY && util.MachineIdentityTokenInstance == nil {
+		// Create new machine identity token instance
+		util.MachineIdentityTokenInstance = util.NewMachineIdentityToken(infisicalMachineIdentityCreds.ClientId, infisicalMachineIdentityCreds.ClientSecret)
+	}
+
+	// TODO: Also save a timestamp of when the token expires, so we know when to refetch an access token
+
+	// if infisicalMachineIdentityCreds.ClientId != "" && infisicalMachineIdentityCreds.ClientSecret != "" {
+	// }
```

Comment:
nit: are these relevant anymore? I know you use the same logic the agent uses now